classes/local/store/s3/client.php

```diff
@@ -635,4 +643,63 @@ public function define_client_check_sdk($config) {
         }
         return $output;
     }
+
+    /**
+     * Proxy range request.
+     *
+     * @param  \stored_file $file    The file to send
+     * @param  object       $ranges  Object with rangefrom, rangeto and length properties.
+     * @return false If couldn't get data.
+     * @throws \coding_exception
+     */
+    public function proxy_range_request(\stored_file $file, $ranges) {
+        // Do not serve files with size less than 2MB or if the feature is disabled.
+        if ($file->get_filesize() < self::MAX_TEMP_LIMIT || empty($this->config->proxyrangerequests)) {
```

Comment:
can you swap the order of this so if the config is off it exist earlier

get_filesize can do some db / api calls in same cases